crates/subspace-farmer/src/farming/tests.rs

```diff
@@ -0,0 +1,118 @@
+use crate::commitments::Commitments;
+use crate::farming::Farming;
+use crate::identity::Identity;
+use crate::mock_rpc::MockRpc;
+use crate::plot::Plot;
+use std::sync::Arc;
+use subspace_core_primitives::{Piece, Salt, Tag, TAG_SIZE};
+use subspace_rpc_primitives::SlotInfo;
+use tempfile::TempDir;
+use tokio::sync::mpsc;
+use tokio::time::{sleep, Duration};
+
+fn init() {
+    let _ = env_logger::builder().is_test(true).try_init();
+}
+
+async fn farming_simulator(slots: Vec<SlotInfo>, tags: Vec<Tag>) {
+    init();
+
+    let base_directory = TempDir::new().unwrap();
+
+    let piece: Piece = [9u8; 4096].into();
+    let salt: Salt = slots[0].salt.clone(); // the first slots salt should be used for the initial commitments
```

Comment:
This is a copy type and some other types below too:
```suggestion
    let salt: Salt = slots[0].salt; // the first slots salt should be used for the initial commitments
```

Reply:
SlotInfo is not `Copy`, hence the compiler gets angry. I know it's component salt is `Copy`. 
```
^^^^^^^^ move occurs because value has type `SlotInfo`, which does not implement the `Copy` trait
```

Reply:
You are not moving `SlotInfo` here, are you sure it is compiler and not IDE being confused?

Reply:
Yes, just tried again. It is compiler

Reply:
I believe the reason is, it's a field of SlotInfo, so it tried to move a part of it, and hence compiler gets angry

Reply:
Doesn't make any sense. Salt is a copy type and when I made this change locally it worked. If the field is a Copy type, it doesn't move anything, it just blindly copies the value. Anyway, not critical.

Reply:
my bad... confused this with another similar-looking line. It works....